File: admin-interface-dotnet/Models/Template.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace admin_interface_dotnet.Models
{
    public class Template
    {
        [Key]
        public Guid Id { get; set; }

        [Required]
        public string UserId { get; set; } = string.Empty;

        [Required]
        [StringLength(254)]
        public string Name { get; set; } = string.Empty;

        public string? Description { get; set; }

        [Required]
        public string Content { get; set; } = string.Empty;

        public List<string> Tags { get; set; } = new List<string>();

        public DateTime CreatedAt { get; set; }
        public DateTime UpdatedAt { get; set; }
        public DateTime? DeletedAt { get; set; }
    }
}

File: src/db/models/Template.ts
import mongoose, { Schema, Document } from 'mongoose';

export interface ITemplate extends Document {
  name: string;
  description?: string;
  category?: string;
  tags?: string[];
  templateData: {
    content: string;
    aiInstructions?: string;
    variables?: Array<{
      name: string;
      type: string;
      required?: boolean;
      description?: string;
      defaultValue?: any;
    }>;
    layout?: {
      pageSize?: string;
      orientation?: string;
      margins?: {
        top?: number;
        bottom?: number;
        left?: number;
        right?: number;
      };
    };
  };
  isActive?: boolean;
  createdAt?: Date;
  updatedAt?: Date;
}

const TemplateSchema = new Schema<ITemplate>({
  name: { type: String, required: true, maxlength: 100 },
  description: { type: String, maxlength: 500 },
  category: { type: String, maxlength: 50 },
  tags: [{ type: String, maxlength: 30 }],
  templateData: {
    content: { type: String, required: true },
    aiInstructions: { type: String },
    variables: [
      {
        name: { type: String, required: true },
        type: { type: String, required: true },
        required: { type: Boolean, default: false },
        description: { type: String },
        defaultValue: { type: Schema.Types.Mixed },
      },
    ],
    layout: {
      pageSize: { type: String, default: 'A4' },
      orientation: { type: String, default: 'portrait' },
      margins: {
        top: { type: Number, default: 20 },
        bottom: { type: Number, default: 20 },
        left: { type: Number, default: 20 },
        right: { type: Number, default: 20 },
      },
    },
  },
  isActive: { type: Boolean, default: true },
}, { timestamps: true });

export const TemplateModel = mongoose.models.Template || mongoose.model<ITemplate>('Template', TemplateSchema);
